schism/video.c

```diff
@@ -276,7 +276,10 @@ void video_startup(void)
 
 void video_resize(unsigned int width, unsigned int height)
 {
-	SDL_RenderSetLogicalSize(video.renderer, width, height);
+	/* Aspect ratio correction if it's wanted */
+	if (cfg_video_want_fixed)
+		SDL_RenderSetLogicalSize(video.renderer, 4, 3);
```

Comment:
Does this actually work correctly in practice with no ill artifacts?

It probably makes more sense to use the NATIVE_SCREEN_{WIDTH,HEIGHT} constants as this API isn't strictly about specifying an aspect ratio AIUI.  It's size *and* ratio.

Reply:
It does work correctly without any artifacts. It works by using the renderer to scale the image, think of it as "post-processing". Other FOSS projects use this, as SDLPoP here:
https://github.com/NagyD/SDLPoP/blob/3e74ba883c72969cbde7037087475b258488245b/src/seg009.c#L2379

As for using NATIVE_SCREEN_{WIDTH,HEIGHT} constants , you're probably right.
I have amended the commit and everything is in place now.

Reply:
It's neat that this works, and looking at SDL2's implementation of SDL_RenderSetLogicalSize, specifically UpdateLogicalSize() @ https://github.com/libsdl-org/SDL/blob/release-2.24.0/src/render/SDL_render.c#L2285 it does appear to strictly consider just the aspect ratio of the logical size operating entirely using floats - in *this* function.

But looking elsewhere in that listing, like the events code @ https://github.com/libsdl-org/SDL/blob/release-2.24.0/src/render/SDL_render.c#L780 it's unclear to me at a glance if this is uniformly working beyond appearances.  Admittedly I'm not familiar with this area of SDL2, and lack the time right now to dig further.

Other considerations like principle of least surprise being if anyone ever starts calling SDL_RenderGetLogicalSize() in schism it's very unlikely they'd expect 4x3 logical pixels to come back.  And yeah, the named symbolic constants of our internal resolution are inherently the aspect ratio defining values, and we should avoid sprinkling unnamed magic numbers throughout the code anyways.

Since you've already changed it to use the constants, this comment is mostly just academic/for posterity.

Reply:
Really, this is the standard way to do aspect ratio correction in SDL2, there's not much more to it.
It works everywhere for what I have seen, this is an stable SDL2 function that shows good results where SDL2 runs.

Reply:
I was going to mention that the native screen constants aren't 4:3, but it looks like you compensated for that already.

I haven't tested this yet, but using `NATIVE_SCREEN_*` instead of `width` and `height` in the `else` case looks wrong to me. IMO better to assume that someone does not want *any* aspect correction if the config var isn't set.

Reply:
Yep, I agree on that. Now using `width` and `height` in the `else`.
No correction means no correction, indeed! :)